src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrItemResourceFactory.java

```diff
@@ -197,4 +201,21 @@ else if (session.itemExists(path)) {
         return item;
     }
 
+    @Nullable Node getParentOrNull(@NotNull Item child, @NotNull String parentPath) {
+        Node parent = null;
+        try {
+            // Use fast getParentOrNull if session is a JackrabbitSession
+            if (this.isJackrabbit) {
+                parent = ((JackrabbitSession) session).getParentOrNull(child);
+            } else if (session.nodeExists(parentPath)) {
+                // Fallback to slower nodeExists & getNode pattern
+                parent = session.getNode(parentPath);
+            }
+        } catch (RepositoryException e) {
+            log.debug("Unable to access node at " + parentPath + ", possibly invalid path", e);
```

Comment:
could this also be because of access control? Would it make sense to mention it as well?

Reply:
most likely it is due to lack of permission.... it's the original message and i didn't alter that one. i would leave out any confusing attempt to explain what could potentially be wrong.... but that's my take.

Reply:
see also line 198

Reply:
hopefully better now :)